packages/moonstone/Image/Image.js

```diff
@@ -62,6 +64,53 @@ const ImageBase = kind({
 	}
 });
 
+
+// This induces a render when there is a screen size change that has a corosponding image src value
+// associated with the new screen size. The render is kicked off by remembering the new image src.
+//
+// This hoc could (should) be rewritten at a later time to use a smarter context API and callbacks,
+// or something like pub/sub; each of which would be hooked together from the resolution.js that
+// would coordinate any screen size/orientation changes and emit events from there.
+//
+// This is ripe for refactoring, and could probably move into UI to be generalized, but that's for
+// another time. -B 2018-05-01
+const ResponsiveImageDecorator = hoc((config, Wrapped) => {
+	return class extends React.Component {
+		static displayName = 'ResponsiveImageDecorator'
+
+		static propTypes = {
+			src: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired
+		}
+
+		constructor (props) {
+			super(props);
+			this.state = {
+				src: selectSrc(this.props.src)
+			};
+		}
+
+		componentDidMount () {
+			window.addEventListener('resize', this.handleResize);
+		}
+
+		componentWillUnmount () {
+			window.removeEventListener('resize', this.handleResize);
+		}
+
+		handleResize = () => {
+			const src = selectSrc(this.props.src);
```

Comment:
Since resize events can happen repeatedly, I suggest rewriting this to use the functional setState.

```js
this.setState((state, props) => {
  const src = selectSrc(props.src);
  if (src != state.src) {
    return {src};
  }

  return null;
});
```

Reply:
I like it. Done.